### PR TITLE
warn about disclosure when using certain options

### PR DESCRIPTION
**Pull request by bcoca**
##### SUMMARY
some options push data inline into cli, this is not good when using secrets and can lead to disclosure 

##### ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request

##### COMPONENT NAME

kubectl

### CODE REVIEW

File: plugins/connection/kubectl.py [hunk 1]
```diff
@@ -65,6 +65,7 @@
       kubectl_extra_args:
         description:
           - Extra arguments to pass to the kubectl command line.
+          - Please be aware that this passes information directly on the command line and it could expose sensitive data.
```
Comment (shanemcd) [inline plugins/connection/kubectl.py]: Should you add the `We recommend using the file based authentication options instead.` here too?

Comment (shanemcd) [reply plugins/connection/kubectl.py]: Guess not? Looks like this was just merged.

Comment (fabianvf) [reply plugins/connection/kubectl.py]: my bad, though I think the password/api key are the only sensitive auth bits you can pass in plain text that you need to work around with file based auth, the other kubectl options are either just paths to files or not sensitive AFAIK

**Comment by bcoca** [comment]
backport https://github.com/ansible/ansible/pull/68195

**Pull request merged**